### PR TITLE
Update dependencies: Use serde_json instead of rustc-serialize, update winapi, log

### PR DESCRIPTION
**Pull request by messense**
[`rustc-serialize`](https://github.com/rust-lang-deprecated/rustc-serialize#rustc-serialize) is deprecated.

### CODE REVIEW

**Comment by messense** [comment]
It seems that Travis CI is disabled?

File: src/json.rs [hunk 1]
```diff
@@ -27,7 +27,7 @@ struct Diagnostic {
     rendered: Option<String>,
```
Comment (messense) [inline src/json.rs]: `rendered` field is unused, should I remove it?

**Comment by laumann** [comment]
Hi @messense Thanks for your PR!

Any chance you could make this PR against rustc/compiletest? This project tries to maintain (some level of) parity with that project, at least to the point of always building on nigthly. New changes in this project should preferably also be pushed back to rustc/compiletest, otherwise it's going to be harder to maintain this project when changes from rustc/compiletest have to be integrated.

> It seems that Travis CI is disabled?

Yes, right now it's just AppVeyor, I haven't really seen the need for both.

**Comment by messense** [comment]
I don't think it's easy to upstream this, according to https://github.com/rust-lang/rust/issues/40527#issuecomment-287230096

I think enable Travis CI is needed to make sure things like`#[cfg(unix)]`(for example [here](https://github.com/laumann/compiletest-rs/blob/6ce0c4a7db88d13df0afaf684bc8490486e89572/src/lib.rs#L19)) annotations actually work and don't break in the future.

**Comment by messense** [comment]
For the maintainability issue, I think this patch is quite small thus should not get in the way of syncing upstream changes to here.

**Comment by laumann** [comment]
> I don't think it's easy to upstream this, according to rust-lang/rust#40527 (comment)

Then I'm reluctant to merge this here, as I'd like to keep the differences between the two projects as minimal as possible. If you look through the history, the most deviation is in `lib.rs` and other changes are predominantly adding things, not changing existing things.

And for the maintainability, every single change looks small, but this kind of thing snowballs quite easily and then becomes a bigger problem than it should be.

@SergioBenitez has mentioned a few times that this project ideally should be rewritten from scratch to address some of the usability issues that arise when compiletest is used outside of rustc. See #81 for an example. This kind of change belongs there in my opinion. But I'd @SergioBenitez's opinion as well - should we merge this?

**Comment by laumann** [comment]
> I think enable Travis CI is needed to make sure things like#[cfg(unix)](for example here) annotations actually work and don't break in the future. cc #97

Again, couldn't this be addressed by adding another target in AppVeyor to have it built on a un*x platform?

**Comment by messense** [comment]
> Again, couldn't this be addressed by adding another target in AppVeyor to have it built on a un*x platform?

Adding another un*x target on AppVeyor would be fine. Although I am not sure how to do that since AppVeyor is a Windows CI service and it will slow down build time since it only have 1 concurrent build.

**Comment by messense** [comment]
https://github.com/rust-lang/rust/pull/48798

It's happening!

**Comment by laumann** [comment]
Yay! Awesome :-) I'll be much happier to merge this then.

**Comment by laumann** [comment]
Oh, rust-lang/rust#48798 was merged, I'll merge this here then.

File: src/json.rs (hunk 1 above)
Comment (laumann) [reply src/json.rs]: I never answered your question here. I'm about to release 0.3.8 and wondering if this warning should be removed?

Comment (laumann) [reply src/json.rs]: I guess it occurs in rustc/compiletest as well?

**Comment by laumann** [comment]
[0.3.8](https://crates.io/crates/compiletest_rs/0.3.8) is out!

File: src/json.rs (hunk 1 above)
Comment (messense) [reply src/json.rs]: https://github.com/rust-lang/rust/blob/master/src/tools/compiletest/src/json.rs#L21-L27

It's gone in rustc compiletest, should be removed.

Comment (laumann) [reply src/json.rs]: Hmmm, ok thanks for checking that, I'll remove it and release 0.3.9 with that if the need arises

**Pull request merged**